### PR TITLE
[Access] Event streaming endpoints

### PR DESCRIPTION
**Pull request by UlyanaAndrukhiv**
https://github.com/onflow/flow-go/issues/5557
 
 ## Description
 This pull request introduces new rpc calls for execution data:

- SubscribeEventsFromStartBlockID - streams events starting at the requested start block id.
- SubscribeEventsFromStartHeight -streams events starting at the requested start block height.
- SubscribeEventsFromLatest  -streams events starting at the latest sealed block.

### CODE REVIEW

File: protobuf/flow/executiondata/executiondata.proto [hunk 1]
```diff
@@ -63,6 +63,55 @@ service ExecutionDataAPI {
   rpc SubscribeEvents(SubscribeEventsRequest)
```
Comment (peterargue) [inline protobuf/flow/executiondata/executiondata.proto:63]: let's add a deprecation notice to this rpc as well

File: protobuf/flow/executiondata/executiondata.proto [hunk 2]
```diff
@@ -63,6 +63,55 @@ service ExecutionDataAPI {
...
+  // Events within each block are filtered by the provided EventFilter, and only
+  // those events that match the filter are returned. If no filter is provided,
+  // all events are returned.
+  //
+  // Responses are returned for each block containing at least one event that
+  // matches the filter. Additionally, heatbeat responses
+  // (SubscribeEventsResponse with no events) are returned periodically to allow
+  // clients to track which blocks were searched. Clients can use this
+  // information to determine which block to start from when reconnecting.
+  rpc SubscribeEventsFromStartBlockID(SubscribeEventsFromStartBlockIDRequest)
```
Comment (peterargue) [inline protobuf/flow/executiondata/executiondata.proto:80]: please also add the docs about errors

File: protobuf/flow/executiondata/executiondata.proto (hunk 1 above)
Comment (UlyanaAndrukhiv) [reply protobuf/flow/executiondata/executiondata.proto:66]: [@d01f7f0](https://github.com/onflow/flow/pull/1441/commits/d01f7f0cf505016e61fdfdf10cf2b49f83fb8eac)

File: protobuf/flow/executiondata/executiondata.proto [hunk 3]
```diff
@@ -63,6 +66,70 @@ service ExecutionDataAPI {
   rpc SubscribeEvents(SubscribeEventsRequest)
       returns (stream SubscribeEventsResponse);
```
Comment (peterargue) [inline protobuf/flow/executiondata/executiondata.proto:67]: I think this is the syntax to mark the RPC deprecated
```suggestion
      returns (stream SubscribeEventsResponse) {
      option deprecated = true;
  };
```

**Review by peterargue — COMMENTED** [review]
added one comment about the deprecation notice, but otherwise looks good

File: protobuf/flow/executiondata/executiondata.proto (hunk 3 above)
Comment (UlyanaAndrukhiv) [reply protobuf/flow/executiondata/executiondata.proto]: [bc18dca](https://github.com/onflow/flow/pull/1441/commits/bc18dca2e2f7ffd0a18b2ea3e3523483b59b1752)

**Review by peterargue — APPROVED** [review]

**Pull request merged**